src/main/java/com/cloudogu/scm/repositorytemplate/RepositoryTemplater.java

```
@@ -0,0 +1,208 @@
+    context.create(filepath).from(templatedContent);
+  }
+
+  private void copyFile(InputStream content, String filepath) throws IOException {
+    context.create(filepath).from(content);
+  }
+
+  private TemplateEngine setupTemplateEngine(RepositoryTemplate repositoryTemplate) {
+    String configEngine = repositoryTemplate.getEngine() != null ? repositoryTemplate.getEngine() : "mustache";
+    TemplateEngine engine = engineFactory.getEngine(configEngine);
+    if (engine == null) {
+      throw NotFoundException.notFound(ContextEntry.ContextBuilder.entity(TemplateEngine.class, configEngine));
+    }
+    return engine;
+  }
+
+  private RepositoryTemplate parseTemplateConfig(RepositoryService repositoryService) throws IOException {
+    Optional<String> templateFile = RepositoryTemplateFinder.templateFileExists(repositoryService);
+    if (!templateFile.isPresent()) {
+      throw NotFoundException.notFound(
+        ContextEntry.ContextBuilder.entity(RepositoryTemplate.class, "template file").in(repositoryService.getRepository())
+      );
+    }
+    try (InputStream templateYml = repositoryService.getCatCommand().getStream(templateFile.get())) {
+      RepositoryTemplate repositoryTemplate = unmarshallRepositoryTemplate(templateFile.get(), templateYml);
+      repositoryTemplate.setNamespaceAndName(repositoryService.getRepository().getNamespaceAndName().toString());
+      if (repositoryTemplate.getFiles() == null) {
+        repositoryTemplate.setFiles(Collections.emptyList());
+      }
+      return repositoryTemplate;
+    }
+  }
+
+  private RepositoryTemplate unmarshallRepositoryTemplate(String templateFile, InputStream templateYml) {
+    RepositoryTemplate repositoryTemplate = yaml.loadAs(templateYml, RepositoryTemplate.class);
+    if (repositoryTemplate == null) {
+      throwTemplateParsingException(templateFile);
+    }
+    return repositoryTemplate;
```

If the `template.yaml` could not be parsed a snakeyaml exception is throw, which leads to an internal server error. We should throw a specific exception with a link to the `template.yaml` syntax documentation.